### PR TITLE
Txif 2998 maddalena/arrow wan2

### PR DESCRIPTION
**Pull request by LenActility**
Driver & model for arrowWan2 from Maddalena

### CODE REVIEW

File: vendors/maddalena/drivers/arrowwan2/index.js [hunk 1]
```diff
@@ -0,0 +1,337 @@
...
+        const buf = Buffer.from(input.bytes, 'hex');
+        const data = decoder.decode(input.fPort, buf);
+        return {data: data.toJSON(), errors: [], warnings: []};
+    } catch (err) {
+        return {errors: [err.message]};
+    }
+}
+
+exports.decodeUplink = decodeUplink;
+/*let input =  {
```
Comment (LenActility) [inline vendors/maddalena/drivers/arrowwan2/index.js:328]: I forgot to delete debug lines

File: vendors/maddalena/models/arrowwan2/model.yaml [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+# Product name & Version
+product:
+    name: MADDALENA/ARROWWAN2
+    version: 1
+# ID(s) of the profile that defines the LoRaWAN characteristics of this model.
+# id: as defined under /profiles.
+# lorawanDeviceProfileID: as defined by LoRa Alliance in https://lora-alliance.org/wp-content/uploads/2020/10/LoRa_Alliance_Vendor_ID_for_QR_Code_02142022.pdf.
+# The lorawanDeviceProfileID consists of 8 upper-case hexadecimal characters, 4 characters for VendorID + 4 characters for VendorProfileID. Please consider using single (') or double (") quotation.
+deviceProfileIds:
+
```
Comment (mostafaibr) [inline vendors/maddalena/models/arrowwan2/model.yaml:16]: Are there some device profiles missing? Strange to add only model without profiles. Is it requested to be used only with a custom device profile?

File: vendors/maddalena/models/arrowwan2/model.yaml [hunk 3]
```diff
@@ -0,0 +1,46 @@
+# Commercial name of the model 
+name: ArrowWan 2
+# Functional description of the product. Maximum 500 characters. 
+description: "Compact wMBus and LoRaWAN™ radio module, ensuring a high level of interoperability with a range of mobile and fixed network reading systems, including third-party ones. Arrow WAN 2 is also suitable for difficult Utility installations (IP68). Activation and configuration of the multiprotocol radio are carried out locally using the NFC interface and a dedicated Android app."
+# Logo of the device
+logo: arrowwan2.png
+# Product name & Version
+product:
+    name: MADDALENA/ARROWWAN2
+    version: 1
```
Comment (mostafaibr) [inline vendors/maddalena/models/arrowwan2/model.yaml:10]:
```suggestion
```

No need for this part in the new device models. It was used before in the already existing devices.

File: vendors/maddalena/drivers/arrowwan2/package.json [hunk 4]
```diff
@@ -0,0 +1,17 @@
+{
+  "name": "maddalena-arrowwan2-driver",
```
Comment (mostafaibr) [inline vendors/maddalena/drivers/arrowwan2/package.json:2]:
```suggestion
  "name": "arrowwan2",
```

**Review by mostafaibr — CHANGES_REQUESTED** [review]

File: vendors/maddalena/drivers/arrowwan2/examples.json [hunk 5]
```diff
@@ -0,0 +1,180 @@
...
+        "meter_serial": "0122123456789",
+        "firmware_version": "1.0",
+        "battery_charge": 99,
+        "pod": "IT 001 E 12345678"
+      },
+      "errors": [],
+      "warnings": []
+    }
+  },
+  {
```
Comment (mostafaibr) [inline vendors/maddalena/drivers/arrowwan2/examples.json:23]: This example failed on unit tests: 

``` 
● Decode uplink › Hourly readings and alarms (two frames for 24 hour values)

    expect(received).toStrictEqual(expected) // deep equality

    - Expected  - 12
    + Received  + 12

    @@ -4,66 +4,66 @@
          "measure_vif": 19,
          "readings": Array [
            Object {
              "delta": 0,
              "reading": 172633,
    -         "timestamp": "2022-05-12T11:00:00",
    +         "timestamp": "2022-05-12T13:00:00",
            },
            Object {
              "delta": 3206,
              "reading": 169427,
    -         "timestamp": "2022-05-12T10:00:00",
    +         "timestamp": "2022-05-12T12:00:00",
            },
            Object {
              "delta": 3171,
              "reading": 166256,
    -         "timestamp": "2022-05-12T09:00:00",
    +         "timestamp": "2022-05-12T11:00:00",
            },
            Object {
              "delta": 3114,
              "reading": 163142,
    -         "timestamp": "2022-05-12T08:00:00",
    +         "timestamp": "2022-05-12T10:00:00",
            },
            Object {
              "delta": 3102,
              "reading": 160040,
    -         "timestamp": "2022-05-12T07:00:00",
    +         "timestamp": "2022-05-12T09:00:00",
            },
            Object {
              "delta": 3118,
              "reading": 156922,
    -         "timestamp": "2022-05-12T06:00:00",
    +         "timestamp": "2022-05-12T08:00:00",
            },
            Object {
              "delta": 3117,
              "reading": 153805,
    -         "timestamp": "2022-05-12T05:00:00",
    +         "timestamp": "2022-05-12T07:00:00",
            },
            Object {
              "delta": 3131,
              "reading": 150674,
    -         "timestamp": "2022-05-12T04:00:00",
    +         "timestamp": "2022-05-12T06:00:00",
            },
            Object {
              "delta": 3131,
              "reading": 147543,
    -         "timestamp": "2022-05-12T03:00:00",
    +         "timestamp": "2022-05-12T05:00:00",
            },
            Object {
              "delta": 3121,
              "reading": 144422,
    -         "timestamp": "2022-05-12T02:00:00",
    +         "timestamp": "2022-05-12T04:00:00",
            },
            Object {
              "delta": 3112,
              "reading": 141310,
    -         "timestamp": "2022-05-12T01:00:00",
    +         "timestamp": "2022-05-12T03:00:00",
            },
            Object {
              "delta": 3115,
              "reading": 138195,
    -         "timestamp": "2022-05-12T00:00:00",
    +         "timestamp": "2022-05-12T02:00:00",
            },
          ],
          "timestamp": "2022-05-13T15:59:30",
          "type": "measure",
        },

      218 |                 adaptDates(result, expected);
      219 |
    > 220 |                 expect(result).toStrictEqual(expected);
          |                                ^
      221 |             });
      222 |         }
      223 |     });

      at Object.toStrictEqual (driver-examples.spec.js:220:32)

```

File: vendors/maddalena/drivers/arrowwan2/examples.json [hunk 6]
```diff
@@ -0,0 +1,180 @@
...
+            "reading": 138195,
+            "delta": 3115
+          }
+        ]
+      },
+      "errors": [],
+      "warnings": []
+    }
+  },
+  {
```
Comment (mostafaibr) [inline vendors/maddalena/drivers/arrowwan2/examples.json:104]: This example failed on unit tests:

```
 ● Decode uplink › Daily readings (7 days) and alarms

    expect(received).toStrictEqual(expected) // deep equality

    - Expected  - 7
    + Received  + 7

    @@ -4,41 +4,41 @@
          "measure_vif": 19,
          "readings": Array [
            Object {
              "delta": 0,
              "reading": 1700,
    -         "timestamp": "2022-06-26T00:00:00",
    +         "timestamp": "2022-06-26T02:00:00",
            },
            Object {
              "delta": 383,
              "reading": 1317,
    -         "timestamp": "2022-06-25T00:00:00",
    +         "timestamp": "2022-06-25T02:00:00",
            },
            Object {
              "delta": 107,
              "reading": 1210,
    -         "timestamp": "2022-06-24T00:00:00",
    +         "timestamp": "2022-06-24T02:00:00",
            },
            Object {
              "delta": -41,
              "reading": 1251,
    -         "timestamp": "2022-06-23T00:00:00",
    +         "timestamp": "2022-06-23T02:00:00",
            },
            Object {
              "delta": 0,
              "reading": 1251,
    -         "timestamp": "2022-06-22T00:00:00",
    +         "timestamp": "2022-06-22T02:00:00",
            },
            Object {
              "delta": 51,
              "reading": 1200,
    -         "timestamp": "2022-06-21T00:00:00",
    +         "timestamp": "2022-06-21T02:00:00",
            },
            Object {
              "delta": 1200,
              "reading": 0,
    -         "timestamp": "2022-06-20T00:00:00",
    +         "timestamp": "2022-06-20T02:00:00",
            },
          ],
          "timestamp": "2022-06-27T00:00:04",
          "type": "measure",
        },

      218 |                 adaptDates(result, expected);
      219 |
    > 220 |                 expect(result).toStrictEqual(expected);
          |                                ^
      221 |             });
      222 |         }
      223 |     });

      at Object.toStrictEqual (driver-examples.spec.js:220:32)

```

**Review by mostafaibr — CHANGES_REQUESTED** [review]

**Review by LenActility — COMMENTED** [review]
I wrote the driver in device catalog private because I didn't know that if we write our proper driver, we do that in the private catalog

**Review by mostafaibr — APPROVED** [review]

**Pull request merged**